chapter04/mdw/Ex-4.pl

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/perl
+
+use 5.010;
+
+use strict;
+
+sub greet{
+
+my $name = shift @_;
+
+state $lastvisitor;
+
+  if ( defined $lastvisitor ){
+     say "Hi $name,  $lastvisitor is also here!";
+     $lastvisitor = $name;
+  } else {
+    say "Hi $name you are the firsts person!";
+    $lastvisitor = $name;
```

Comment:
instead of duplicating the $lastvisitor = $name line, you could put it once just after the if